src/SprykerSdk/Zed/AppSdk/Communication/Console/AppConfigurationCreateConsole.php

```diff
@@ -0,0 +1,595 @@
+    {
+        do {
+            $propertyName = $this->setPropertyName($this->askTextQuestion($input, $output, $this->getPropertyNameQuestion()));
+
+            $this->setIsRequired(
+                $propertyName,
+                $this->askChoiceQuestion($input, $output, $this->getIsRequiredQuestion(), $this->getConfirmationOptions(), 0),
+            );
+
+            $this->setWidget(
+                $input,
+                $output,
+                $propertyName,
+                $this->askChoiceQuestion($input, $output, $this->getWidgetQuestion(), $this->getWidgetOptions(), 0),
+            );
+        } while ($this->askChoiceQuestion($input, $output, $this->getNewConfigurationConfirmation(), $this->getConfirmationOptions(), 0) == 'Yes');
+    }
+
+    /**
+     * @return array
+     */
+    protected function getConfirmationOptions(): array
+    {
+        return ['Yes', 'No'];
+    }
+
+    /**
+     * @return string
+     */
+    protected function getPropertyNameQuestion(): string
+    {
+        return 'Please enter a name: ';
+    }
+
+    /**
+     * @param string $propertyName
+     *
+     * @return string
+     */
+    protected function setPropertyName(string $propertyName): string
+    {
+        $this->properties[$propertyName] = ['placeholder' => $propertyName];
+
+        return $propertyName;
+    }
+
+    /**
+     * @return string
+     */
+    protected function getIsRequiredQuestion(): string
+    {
+        return 'Is this a required field?';
+    }
+
+    /**
+     * @param string $propertyName
+     * @param string $required
+     *
+     * @return void
+     */
+    protected function setIsRequired(string $propertyName, string $required): void
+    {
+        $this->properties[$propertyName]['isRequired'] = ($required == 'Yes') ? true : false;
+        $this->required[] = $propertyName;
+    }
+
+    /**
+     * @return string
+     */
+    protected function getWidgetQuestion(): string
+    {
+        return 'Please select a widget: ';
+    }
+
+    /**
+     * @return string
+     */
+    protected function getMoreOptionConfirmation(): string
+    {
+        return 'Do you want to add more options?';
+    }
+
+    /**
+     * @return string
+     */
+    protected function getNewConfigurationConfirmation(): string
+    {
+        return 'Do you want to add more configurations?';
+    }
+
+    /**
+     * @return array
+     */
+    protected function getWidgetOptions(): array
+    {
+        return ['Text', 'Radio', 'Checkbox'];
+    }
+
+    /**
+     * @param \Symfony\Component\Console\Input\InputInterface $input
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
+     * @param string $propertyName
+     * @param string $widget
+     *
+     * @return void
+     */
+    protected function setWidget(InputInterface $input, OutputInterface $output, string $propertyName, string $widget): void
+    {
+        $this->properties[$propertyName]['widget']['id'] = $widget;
```

Comment:
Please only collect information in the console and construct the file content in the business layer.

When we want to switch the content of the configuration file are use a completely different one we would need to change the whole class. The purpose of the Console commands is to collect information and pass it to the business layer.

The business layer decides how to handle the data and creates the file.

Now *this* class is a mixture of communication and business.

Reply:
I made changed and pushed again. Please check and let me know your feedback.